From 67a270ec7c07baf3b1ec9e97ff868ecc1bca5ea8 Mon Sep 17 00:00:00 2001
From: Adrian Alonso <adrian.alonso@nxp.com>
Date: Mon, 25 Jun 2018 15:06:32 +0900
Subject: [PATCH 4073/5242] MLK-18704-3: ARM64: dts: freescale: imx8mm evk:
 enable pwm4 led

commit  b678ee0c59b741f0246b000634cdbdee7f06c50e from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable pwm4 control on status led on imx8mm evk base board
use hartbeat trigger to blink led

Signed-off-by: Adrian Alonso <adrian.alonso@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
(cherry picked from commit 65b2de72509084a577e37b2758061612a3c7d639)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts |   23 ++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
index bb074d7..ed489e9 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
@@ -45,6 +45,17 @@
 		#reset-cells = <0>;
 	};
 
+	pwmleds {
+		compatible = "pwm-leds";
+		ledpwm4 {
+			label = "PWM4";
+			pwms = <&pwm4 0 50000>;
+			max-brightness = <255>;
+			linux,default-trigger = "heartbeat";
+			active-low;
+		};
+	};
+
 	regulators {
 		compatible = "simple-bus";
 		#address-cells = <1>;
@@ -215,6 +226,12 @@
 			>;
 		};
 
+		pinctrl_pwm4: pwm4grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_GPIO1_IO15_PWM4_OUT	0x16
+			>;
+		};
+
 		pinctrl_pmic: pmicirq {
 			fsl,pins = <
 				MX8MM_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x41
@@ -860,6 +877,12 @@
 	status = "okay";
 };
 
+&pwm4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm4>;
+	status = "okay";
+};
+
 &uart1 { /* BT */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1>;
-- 
1.7.9.5

